File: dotnet/PersonasABM.Domain/Entities/AtributoTipo.cs
using System.ComponentModel.DataAnnotations;

namespace PersonasABM.Domain.Entities;

public class AtributoTipo
{
    public int Id { get; set; }
    
    [Required]
    [StringLength(100)]
    public string Nombre { get; set; } = string.Empty;
    
    [StringLength(500)]
    public string? Descripcion { get; set; }
    
    [Required]
    [StringLength(20)]
    public string TipoDato { get; set; } = "Texto"; // Texto, Booleano, Numero, Fecha
    
    public bool EsObligatorio { get; set; } = false;
    
    public bool Activo { get; set; } = true;
    
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    
    // Navegación
    public virtual ICollection<PersonaAtributo> PersonaAtributos { get; set; } = new List<PersonaAtributo>();
}

File: nodejs/src/entities/AtributoTipo.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  OneToMany,
} from 'typeorm';
import { PersonaAtributo } from './PersonaAtributo';

@Entity('atributo_tipos')
export class AtributoTipo {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ type: 'varchar', length: 100 })
  nombre!: string;

  @Column({ type: 'varchar', length: 500, nullable: true })
  descripcion?: string;

  @Column({ type: 'varchar', length: 20, default: 'Texto' })
  tipoDato!: string;

  @Column({ type: 'boolean', default: false })
  esObligatorio!: boolean;

  @Column({ type: 'boolean', default: true })
  activo!: boolean;

  @CreateDateColumn()
  fechaCreacion!: Date;

  @OneToMany(
    () => PersonaAtributo,
    (personaAtributo) => personaAtributo.atributoTipo
  )
  personaAtributos!: PersonaAtributo[];
}
